feature-libs/product-multi-dimensional/selector/core/services/product-multi-dimensional-selector-images.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+        },
+        {
+          image: { format: 'large', url: '/image2.jpg' },
+        },
+      ];
+      const variantOptionValue = 'Sample Variant';
+
+      const result = service.getVariantOptionImages(
+        variantOptionQualifiers,
+        variantOptionValue
+      );
+
+      expect(result.length).toBe(1);
+      expect(result[0]).toEqual({
+        altText: variantOptionValue,
+        format: VariantQualifier.STYLE_SWATCH,
+        url: 'http://media.base.url//image1.jpg',
+      });
+    });
+
+    it('should use altText from image if available', () => {
+      config.multiDimensional.imageFormat = VariantQualifier.STYLE_SWATCH;
+      const variantOptionQualifiers: VariantOptionQualifier[] = [
+        {
+          image: {
+            format: VariantQualifier.STYLE_SWATCH,
+            url: '/image1.jpg',
+            altText: 'Alt Text',
+          },
+        },
+      ];
+      const variantOptionValue = 'Sample Variant';
+
+      const result = service.getVariantOptionImages(
+        variantOptionQualifiers,
+        variantOptionValue
+      );
+
+      expect(result.length).toBe(1);
+      expect(result[0]).toEqual({
+        altText: 'Alt Text',
+        format: VariantQualifier.STYLE_SWATCH,
+        url: 'http://media.base.url//image1.jpg',
+      });
+    });
+
+    it('should handle missing config gracefully', () => {
+      config.multiDimensional = undefined;
+      const variantOptionQualifiers: VariantOptionQualifier[] = [
+        {
+          image: { format: VariantQualifier.STYLE_SWATCH, url: '/image1.jpg' },
+        },
+      ];
+      const variantOptionValue = 'Sample Variant';
+
+      const result = service.getVariantOptionImages(
+        variantOptionQualifiers,
+        variantOptionValue
+      );
+
+      expect(result.length).toBe(0);
+    });
+
+    it('should handle missing image properties gracefully', () => {
+      config.multiDimensional.imageFormat = VariantQualifier.STYLE_SWATCH;
+      const variantOptionQualifiers: VariantOptionQualifier[] = [
+        {
+          image: undefined,
+        },
+      ];
+      const variantOptionValue = 'Sample Variant';
+
+      const result = service.getVariantOptionImages(
+        variantOptionQualifiers,
+        variantOptionValue
+      );
+
+      expect(result.length).toBe(0);
+    });
+  });
+
+  describe('getBaseUrl', () => {
+    it('should return media baseUrl if available', () => {
+      const result = service['getBaseUrl']();
+      expect(result).toBe('http://media.base.url/');
+    });
+
+    it('should return occ baseUrl if media baseUrl is not available', () => {
+      config.backend.media.baseUrl = undefined;
+
+      const result = service['getBaseUrl']();
+      expect(result).toBe('http://occ.base.url/');
+    });
+
+    it('should return empty string if no baseUrl is available', () => {
+      config.backend.media.baseUrl = undefined;
+      config.backend.occ.baseUrl = undefined;
```

Comment:
```suggestion
      config.backend = {
        media: { baseUrl: undefined },
        occ: { baseUrl: undefined },
      };
```
to avoid TS errors

Reply:
fixed